vms/evm/database/wrapped_database.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"errors"
+
+	"github.com/ava-labs/libevm/ethdb"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+var (
+	_ ethdb.Batch         = (*ethBatchWrapper)(nil)
+	_ ethdb.KeyValueStore = (*ethDBWrapper)(nil)
+
+	ErrSnapshotNotSupported = errors.New("snapshot is not supported")
+	ErrStatNotSupported     = errors.New("stat is not supported")
+)
+
+func WrapDatabase(db database.Database) ethdb.KeyValueStore { return ethDBWrapper{db} }
+
+type ethDBWrapper struct{ database.Database }
```

Comment:
1. nit: I would just call this `database` or `db` or something. It's implied that this is an eth-compatible db because we're in the `evm` package. It's also obvious from looking at the struct definition that this is a wrapper pattern.
2. avoid embed if we can, see later comment

Reply:
1. We can't do `database` (it's the import`) or `db` bc of the intervening var definition. I'd say we do `ethdb`, but that's also an import. Any other suggestions?

Reply:
1. You can alias imports if there's a namespace collision (you could call `database` `avalanchegodatabase` or `avalanchegodb` or something)